corim/example_profile_test.go

```diff
@@ -0,0 +1,199 @@
+// constraints:
+// - Entities may contain an address field
+// - Reference values may contain a Unix timestamp indicating when the
+//   individual measurement was taken.
+// - The language claim (CoMID index 0) must be present, and must be "en-GB"
+
+// EntityExtensions will be used for both CoMID and CoRIM entities.
+type EntityExtensions struct {
+	Address *string `cbor:"-1,keyasint,omitempty" json:"address,omitempty"`
+}
+
+type RefValExtensions struct {
+	Timestamp *int `cbor:"-1,keyasint,omitempty" json:"timestamp,omitempty"`
+}
+
+// We're not defining any additional fields, however we're providing extra
+// constraints that will be applied on top of standard CoMID validation.
+type ComidExtensions struct{}
+
+func (*ComidExtensions) ConstrainComid(c *comid.Comid) error {
+	if c.Language == nil {
+		return errors.New("language not specified")
+	}
+
+	if *c.Language != "en-GB" {
+		return fmt.Errorf(`language must be "en-GB", but found %q`, *c.Language)
+	}
+
+	return nil
+}
+
+// Registering the profile inside init() in the same file where it is defined
+// ensures that the profile will always be available, and you don't need to
+// remember to register it at the time you want to use it. The only potential
+// danger with that is if the your profile ID clashes with another profile,
+// which should not happen if it a registered PEN or a URL containing a domain
+// that you own.
+func init() {
+	profileID, err := eat.NewProfile("http://example.com/example-profile")
+	if err != nil {
+		panic(err) // will not error, as the hard-coded string above is valid
+	}
+
+	extMap := extensions.NewMap().
+		Add(ExtEntity, &EntityExtensions{}).
+		Add(comid.ExtComid, &ComidExtensions{}).
+		Add(comid.ExtEntity, &EntityExtensions{}).
+		Add(comid.ExtReferenceValue, &RefValExtensions{})
+
+	if err := RegisterProfile(profileID, extMap); err != nil {
+		// will not error, assuming our profile ID is unique, and we've
+		// correctly set up the extensions Map above
+		panic(err)
+	}
+}
+
+// ----- end of profile definition -----
+// The following code demonstrates how the profile might be used.
+
+func Example_profile_unmarshal() {
+	buf, err := os.ReadFile("testcases/unsigned-example-corim.cbor")
+	if err != nil {
+		log.Fatalf("could not read corim file: %v", err)
+	}
+
+	// UnmarshalUnsignedCorimFromCBOR will detect the profile and ensure
+	// the correct extensions are loaded before unmarshalling
+	extractedCorim, err := UnmarshalUnsignedCorimFromCBOR(buf)
+	if err != nil {
+		log.Fatalf("could not unmarshal corim: %v", err)
+	}
+
+	extractedComid, err := UnmarshalComidFromCBOR(
+		extractedCorim.Tags[0],
+		extractedCorim.Profile,
+	)
+	if err != nil {
+		log.Fatalf("could not unmarshal corim: %v", err)
+	}
+
+	fmt.Printf("Language: %s\n", *extractedComid.Language)
+	fmt.Printf("Entity: %s\n", *extractedComid.Entities.Values[0].Name)
+	fmt.Printf("        %s\n", extractedComid.Entities.Values[0].
```

Comment:
```suggestion
	fmt.Printf(" Address:       %s\n", extractedComid.Entities.Values[0].
```

Reply:
The address is part of the Entity, not a sparte entry.